x-pack/plugins/stack_alerts/server/alert_types/geo_containment/es_query_builder.ts

```diff
@@ -53,28 +52,40 @@ export async function getShapesFilters(
     index: boundaryIndexTitle,
     body: {
       size: MAX_SHAPES_QUERY_SIZE,
+      _source: false,
+      fields: boundaryNameField ? [boundaryNameField] : [],
       ...(boundaryIndexQuery ? { query: getEsFormattedQuery(boundaryIndexQuery) } : {}),
     },
   });
 
-  boundaryData.hits.hits.forEach(({ _index, _id }: { _index: string; _id: string }) => {
-    filters[_id] = {
+  for (let i = 0; i < boundaryData.hits.hits.length; i++) {
```

Comment:
Query results processing optimization - tight loops with large sizes should not use forEach since that creates objects for each iteration and increases pressure on garbage collection.